AutoGraph/ResponseHandler.swift

```diff
@@ -40,16 +40,28 @@ open class ResponseHandler {
     private func map<SerializedObject: Decodable>(json: JSONValue, response: HTTPURLResponse?, objectBinding: ObjectBinding<SerializedObject>) {
             do {
                 switch objectBinding {
-                case .object(let keyPath, let isRequestIncludingJSON, let completion):
+                case .object(let keyPath, let isRequestIncludingNetworkResponse, let completion):
                     let decoder = JSONDecoder()
                     let decodingJSON: JSONValue = try {
                         guard let objectJson = json[keyPath] else {
                             throw ObjectKeyPathError(keyPath: keyPath)
                         }
                         
-                        switch isRequestIncludingJSON {
-                        case true:  return .object(["json" : objectJson, "value" : objectJson])
-                        case false: return objectJson
+                        switch isRequestIncludingNetworkResponse {
+                        case true:
+                            var result: [String : JSONValue] = ["json" : objectJson, "value" : objectJson]
+                            if let httpResponse = response {
+                                let http = try? JSONValue(dict: [
+                                    "urlString" : httpResponse.url?.absoluteString ?? "",
+                                    "statusCode" : httpResponse.statusCode,
+                                    "headerFields" : httpResponse.allHeaderFields
+                                ])
+                                result["httpResponse"] = http
+                            }
```

Comment:
could invert this and result to keep things immutable.
```swift
let httpResponseJson = response.map {
    try? JSONValue(dict: [
        "urlString" : $0.url?.absoluteString ?? "",
        "statusCode" : $0.statusCode,
        "headerFields" : $0.allHeaderFields
    ])
} ?? .null

let result: [String : JSONValue] = [
    "json" : objectJson,
    "value" : objectJson,
    "httpResponse" : httpResponseJson
]
```
This also clues the ultimate receiver of this data into the fact that `httpResponse` is null, which frankly, should always be there in our setup anyway and shouldn't happen.

Reply:
Nice.